File: src/main/java/org/sgambe/wtplotter/replaydata/CraftInfo.java
package org.sgambe.wtplotter.replaydata;

import com.google.gson.JsonObject;

public class CraftInfo {
    private String name;
    private String type;
    private boolean rankUnused;
    private int mrank;
    private int rank;


    public CraftInfo fromJson(JsonObject craftInfoObject) {
        this.name = craftInfoObject.get("name").getAsString();
        this.type = craftInfoObject.get("type").getAsString();
        this.rankUnused = craftInfoObject.get("rankUnused").getAsBoolean();
        this.mrank = craftInfoObject.get("mrank").getAsInt();
        this.rank = craftInfoObject.get("rank").getAsInt();
        return this;

    }

    public int getRank() {
        return rank;
    }

    public int getMrank() {
        return mrank;
    }

    public boolean isRankUnused() {
        return rankUnused;
    }

    public String getType() {
        return type;
    }

    public String getName() {
        return name;
    }

    @Override
    public String toString() {
        return "CraftInfo{" + "name='" + name + '\'' + ", type='" + type + '\'' + ", rankUnused=" + rankUnused + ", mrank=" + mrank + ", rank=" + rank + '}';
    }
}

File: classes/craftinfo.cpp
#include "craftinfo.h"

CraftInfo::CraftInfo() {}

CraftInfo CraftInfo::fromJson(const QJsonObject& craftInfoObject) {
	CraftInfo craftInfo;

	if (craftInfoObject.contains("name") && craftInfoObject["name"].isString()) {
		craftInfo.name = craftInfoObject["name"].toString();
	}

	if (craftInfoObject.contains("type") && craftInfoObject["type"].isString()) {
		craftInfo.type = craftInfoObject["type"].toString();
	}

	if (craftInfoObject.contains("rankUnused") && craftInfoObject["rankUnused"].isBool()) {
		craftInfo.rankUnused = craftInfoObject["rankUnused"].toBool();
	}

	if (craftInfoObject.contains("mrank") && craftInfoObject["mrank"].isDouble()) {
		craftInfo.mrank = craftInfoObject["mrank"].toInt();
	}

	if (craftInfoObject.contains("rank") && craftInfoObject["rank"].isDouble()) {
		craftInfo.rank = craftInfoObject["rank"].toInt();
	}

	return craftInfo;
}

int CraftInfo::getRank() {
	return rank;
}

int CraftInfo::getMrank() {
	return mrank;
}

bool CraftInfo::isRankUnused() {
	return rankUnused;
}

QString CraftInfo::getType() {
	return type;
}

QString CraftInfo::getName() {
	return name;
}



